src/lib/comms/sol-http-client-impl-curl.c

```diff
@@ -938,3 +995,25 @@ sol_http_client_connection_cancel(struct sol_http_client_connection *pending)
     SOL_INT_CHECK(sol_ptr_vector_remove(&global.connections, pending), < 0);
     destroy_connection(pending);
 }
+
+SOL_API struct sol_http_client_connection *
+sol_http_client_request_with_interface(enum sol_http_method method,
+    const char *base_uri, const struct sol_http_params *params,
+    const struct sol_http_request_interface *interface,
+    const void *data)
+{
+    struct sol_http_request_interface *iface;
+    struct sol_http_client_connection *pending;
+
+    iface = sol_util_memdup(interface, sizeof(*iface));
```

Comment:
btw, the struct should be versioned and the version checked here if compiled with `api_version` support.